### DISCUSSION TITLE
Issue deploying Edge Function

### DISCUSSION THREAD

**Discussion by tsetserra2024** [supabase/supabase#28645] [Questions] [answered]
I have the following edge function:

> import { serve } from 'https://deno.land/x/sift@0.4.3/mod.ts'
> import { createClient } from 'https://deno.land/x/supabase@1.3.1/mod.ts'
> 
> // Initialize Supabase client
> const supabase = createClient(Deno.env.get('PROJECT_SUPABASE_URL')!, Deno.env.get('PROJECT_SUPABASE_SERVICE_ROLE_KEY')!)
> 
> // Google API client setup
> const serviceAccount = JSON.parse(Deno.env.get('GOOGLE_SERVICE_ACCOUNT_JSON')!)
> import { google } from 'https://deno.land/x/googleapis@v87.0.0/googleapis.ts'
> 
> async function getUsersAndInsert(delegatedUserEmail: string, primaryOwnerUserId: string) {
>   try {
>     // Authenticate with Google API, impersonating the delegated user
>     const auth = new google.auth.GoogleAuth({
>       credentials: serviceAccount,
>       scopes: [
>         'https://www.googleapis.com/auth/admin.directory.user.readonly',
>       ],
>     });
> 
>     const delegatedAuth = await auth.getClient({ subject: delegatedUserEmail });
>     const admin = google.admin({ version: 'directory_v1', auth: delegatedAuth });
> 
>     // Retrieve users from the specified domain
>     const res = await admin.users.list({ customer: 'my_customer' });
>     const users = res.data.users || [];
> 
>     // Get the current timestamp for created_at field
>     const createdAt = new Date().toISOString();
> 
>     // Map users to the Supabase insertion format
>     const userRecords = users.map(user => ({
>       primary_owner_user_id: primaryOwnerUserId,
>       name: `${user.name?.givenName || ''} ${user.name?.familyName || ''}`.trim(),
>       email: user.primaryEmail || '',
>       is_personal_user: true,
>       admingoogleconfirmation: false,
>       superadmingoogleconfirmation: false,
>       created_at: createdAt
>     }));
> 
>     // Insert each user into the Supabase 'users' table
>     for (const record of userRecords) {
>       await supabase
>         .from('users')
>         .insert([record]);
>     }
> 
>     return new Response('Data inserted successfully', { status: 200 });
>   } catch (error) {
>     console.error('Error fetching users or inserting into Supabase:', error);
>     return new Response('Error inserting data', { status: 500 });
>   }
> }
> 
> // Main function to serve requests
> serve(async (req) => {
>   const url = new URL(req.url);
>   
>   // Extract parameters from the request
>   const delegatedUserEmail = url.searchParams.get('delegatedUserEmail');
>   const primaryOwnerUserId = url.searchParams.get('primaryOwnerUserId');
> 
>   if (!delegatedUserEmail || !primaryOwnerUserId) {
>     return new Response('Missing required parameters', { status: 400 });
>   }
> 
>   return await getUsersAndInsert(delegatedUserEmail, primaryOwnerUserId);
> });

When running 'supabase functions deploy getUsersAndInsert' I get the following error message:
`Error: failed to load 'https://x.lcas.dev/preact@10.5.12/ssr.js': Import 'https://x.lcas.dev/preact@10.5.12/ssr.js' failed: error sending request for url (https://x.lcas.dev/preact@10.5.12/ssr.js): error trying to connect: dns error: failed to lookup address information: Name or service not known
2024/08/14 09:35:25 Sent Header: Host [api.moby.localhost]
2024/08/14 09:35:25 Sent Header: User-Agent [Docker-Client/unknown-version (linux)]
2024/08/14 09:35:25 Send Done
2024/08/14 09:35:25 Recv First Byte
error running container: exit 1`

**Comment by joeeardley** [comment]
I've got the same issue as well - the x.lcas.dev domain seems to be down. Sounds like a potential workaround and saving the ssr.js file locally and changing the import to point at the local file but I can't find a copy of the ssr.js file anywhere

**Comment by tsetserra2024** [comment, marked as answer]
I found by changing out: import { serve } from 'https://deno.land/x/sift@0.4.3/mod.ts'
For this: import { serve } from 'https://deno.land/x/sift@0.6.0/mod.ts'

This resolved this specific error message. 
This new mod.ts file doesn't have the reference to the ssr.js file endpoint that is currently down.

**Reply by erkinovalim** [reply]
Thank you it worked for me!
